deepchem/models/tensorgraph/models/chemnet_models.py

```diff
@@ -0,0 +1,366 @@
+"""
+Smiles2Vec model, described in https://arxiv.org/pdf/1712.02034.pdf
+ChemCeption model, described in https://arxiv.org/pdf/1706.06689.pdf
+"""
+
+from __future__ import division
+from __future__ import unicode_literals
+
+__author__ = "Vignesh Ram Somnath"
+__license__ = "MIT"
+
+import numpy as np
+import tensorflow as tf
+import os
+import sys
+import logging
+
+from deepchem.data.datasets import pad_batch
+from deepchem.models import KerasModel, layers
+from deepchem.models.losses import L2Loss, SoftmaxCrossEntropy
+from deepchem.metrics import to_one_hot
+from deepchem.models.tensorgraph.layers import KerasLayer
+from deepchem.models.tensorgraph import chemnet_layers
+from tensorflow.keras.layers import Input, Dense, Reshape, Softmax
+from tensorflow.keras.layers import Dropout, Conv1D, Concatenate, Lambda, GRU, LSTM, Bidirectional
+from tensorflow.keras.layers import Conv2D, ReLU, Add, GlobalAveragePooling2D
+
+DEFAULT_INCEPTION_BLOCKS = {"A": 3, "B": 3, "C": 3}
+
+INCEPTION_DICT = {
+    "A": chemnet_layers.InceptionResnetA,
+    "B": chemnet_layers.InceptionResnetB,
+    "C": chemnet_layers.InceptionResnetC
+}
+
+RNN_DICT = {"GRU": GRU, "LSTM": LSTM}
+
+
+class Smiles2Vec(KerasModel):
+  """
+  Smiles2Vec is a RNN basel model that ingests SMILES characters,
+  get their embeddings. These embeddings are fed through a series of RNN
+  (and optionally Conv) layers, with the representations from the final layer
+  as input for classification or regression tasks.
+  """
+
+  def __init__(self,
+               char_to_idx,
+               n_tasks=10,
+               batch_size=64,
+               max_seq_len=270,
+               embedding_dim=50,
+               n_classes=2,
+               use_bidir=True,
+               use_conv=True,
+               filters=192,
+               kernel_size=3,
+               strides=1,
+               rnn_sizes=[224, 384],
+               rnn_types=["GRU", "GRU"],
+               learning_rate=0.0001,
+               model_dir=None,
+               tensorboard=True,
+               optimizer=None,
+               mode="regression",
+               tensorboard_log_frequency=5,
+               restore_from=None,
+               **kwargs):
```

Comment:
You don't need to list all the keyword arguments from the parent class.  They'll already get passed along through `kwargs`.  Doing it that way has two big advantages.  First, it avoids bug opportunities (not passing one of the arguments on to the parent class).  Second, if we change some of the parent class arguments in the future, we only have to change them in one place rather than in every subclass.

Reply:
I see! I will change this too.

Reply:
Done.